text_handler.py

```diff
@@ -1,86 +1,76 @@
         try:
-            yt = YouTube(message.text)
-            author = yt.author
-            length = yt.length
-            title = yt.title
-            stream = (yt.streams
-                      .filter(only_audio=True, file_extension='mp4')
-                      .order_by('bitrate').desc().first())
-            filesize = stream.filesize
-            video_id = extract.video_id(message.text)
-            video_url = "youtu.be/" + video_id
-            telegram_audio_limit = 52428800
-
-            if (filesize < telegram_audio_limit and length < 600):
-                output_audiofile = stream.download(output_path="downloads/",
-                                                   filename=title +
-                                                   " - YouTube - " +
-                                                   video_id)
-                output_thumbnail = "downloads/square_thumbnail_" + video_id + \
-                                   ".jpg"
-                self.__make_squarethumb(yt.thumbnail_url, output_thumbnail)
-                with open(output_audiofile, "rb") as audio, \
-                     open(output_thumbnail, "rb") as thumb:
-                    await message.reply_audio(audio,
-                                              caption="<a href=\"" + video_url +
-                                                      "\">" + title + "</a>",
-                                              parse_mode=ParseMode.HTML,
-                                              duration=length,
-                                              performer=author,
-                                              title=title,
-                                              thumb=thumb)
-                os.remove(output_audiofile)
-                os.remove(output_thumbnail)
-            else:
-                await message.reply("`No downloads for 10min\\+ audio " +
-                                    "or file size greater than 50M`",
-                                    parse_mode=ParseMode.MARKDOWN_V2)
+            info = musicDownloader.download(message.text,
+                                            telegram_audio_limit)
+            downloads = info['downloads']
+            audio_file = downloads['audio']
+            thumbnail_file = downloads['thumbnail']
+            squarethumb_file = DOWNLOAD_DIR + \
+                info['extractor'] + info['id'] + "_squarethumb.jpg"
+            squarethumbMaker.make_squarethumb(thumbnail_file,
+                                              squarethumb_file)
+            with open(audio_file, "rb") as audio, \
+                 open(squarethumb_file, "rb") as thumb:
+                title = info['title']
+                url = info['webpage_url']
+                performer = info['uploader']
+                duration = int(float(info['duration']))
+                caption = "<b><a href=\"" + url + "\">" + title + "</a></b>"
+                await types.ChatActions.upload_document()
```

Comment:
so this upload_document() action gets canceled automatically when the bot sends a message?

Reply:
Yes, it won't keep showing that status.
> The status is set for 5 seconds or less (when a message arrives from your bot, Telegram clients clear its typing status)
https://core.telegram.org/bots/api#user